### PR TITLE
Allow `hardware.place_order` with size ID (and name)

### PR DESCRIPTION
**Pull request by ZuluPro**

### CODE REVIEW

**Comment by allmightyspiff** [comment]
The CI check should be fixed in the latest master commit. 

Also, are there any presets that can't be ordered here with just their size name?

**Comment by kwienken** [comment]
@allmightyspiff I do not see anything here that would cause problems with ordering a preset. 

I'm not thrilled with allowing an id to be used instead of the size (keyName) since we tend to encourage keyNames over ids. I guess it's not that much of a risk though since a preset id isn't as likely to change as something like a price id.

**Comment by ZuluPro** [comment]
Hello,@kwienken
In fact, I'm using ID because (default) hardware listing doesn't give the preset's name but its ID.

**Comment by kwienken** [comment]
@ZuluPro in that case I'm fine with using the ID instead of keyName.

File: SoftLayer/managers/hardware.py [hunk 1]
```diff
@@ -467,13 +467,18 @@ def _generate_create_dict(self,
             'hostname': hostname,
             'domain': domain,
         }
-
+        try:
+            size = _get_preset_id(package, size)
+        except SoftLayer.SoftLayerError as err:
+            if not err.message.startswith('Could not find valid size for'):
+                raise
+            size = size
```
Comment (ZuluPro) [inline SoftLayer/managers/hardware.py]: As you asked I check if a size exists with its name before take ID.

I check with try/except, and raise other exceptions.

**Comment by ZuluPro** [comment]
@allmightyspiff @kwienken 
Any news ?

**Comment by ZuluPro** [comment]
Any news ?

**Comment by allmightyspiff** [comment]
I'm going to address this in #912

**Pull request closed without merging**